Cleanup OvrIntegration issues

Squareys opened the pull request:
Hi @mosra !

I finally got around to cleaning up #13... sorry it took me so long. And sorry for the mess, some of the things I fixed were rather sloppy.

Regards, Jonathan.

PS: Take your time with the review, I will need to compile check the example still anyway, and this shouldn't be urgent in any way.

mosra commented on src/Magnum/OvrIntegration/Compositor.cpp:
```diff
@@ -108,8 +103,6 @@ LayerQuad& LayerQuad::setQuadSize(const Vector2& size) {
     return *this;
 }
 
-Compositor::Compositor() = default;
```
I would keep the constructor here, the class has heavy members (`std::vector`) and having the constructor inlined will increase compile time and binary size.

mosra commented on src/Magnum/OvrIntegration/Context.cpp:
```diff
@@ -80,4 +80,15 @@ std::unique_ptr<Session> Context::createSession() {
+
+    Error err;
+    err.type = ErrorType(info.Result);
+    strncpy(err.message, info.ErrorString, 512);
```
`std::strncpy()`?

mosra commented:
Thanks a lot! :+1: 

I still saw mentions of the OVR API instead of the wrapped API in the error enum, can you look there also?

Squareys commented:
@mosra I addressed the new issues (de-inlined the Compositor constructor, referenced OvrIntegration types in errors enum and added the `std::`). This could possibly be good to go now :)

Regards, Squareys.

Squareys commented:
@mosra Rebased and removed error enum values to know be OculusSDK 1.6.0 compliant!

I would be really happy, if this could get merged in the next 7 days, if that is possible somehow :)

mosra commented on src/Magnum/OvrIntegration/Session.cpp:
```diff
@@ -37,6 +37,29 @@
+
+    /** The HMD was created as a debug HMD (without real hardware) */
+    Debug = ovrHmdCap_DebugDevice /* 0x0010 */
+};
```
:+1: for the idea of having the enum definition only in the `*.cpp` file. A new thing for me! :)

mosra commented:
Merged. Thanks a lot!
